Makefile

```diff
@@ -160,3 +128,88 @@ golangci-lint-fix.%: $(GOLANGCI_LINT)
 
 .PHONY: FORCE # use this to force phony behavior for targets with pattern rules
 FORCE:
+
+##@ Deployment
+
+.PHONY: cluster
+cluster: kind ctlptl ## Create Kind cluster and local registry
+	$(CTLPTL) apply -f ctlptl.yaml
+
+.PHONY: cluster-reset
+cluster-reset: kind ctlptl ## Delete Kind cluster
+	$(CTLPTL) delete -f ctlptl.yaml
+
+ifndef ignore-not-found
+  ignore-not-found = false
+endif
+
+.PHONY: deploy
+deploy: .gen kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	$(KUSTOMIZE) build . | $(KUBECTL) apply -f -
+
+.PHONY: undeploy
+undeploy: kustomize ## Undeploy controller from the K8s cluster specified in ~/.kube/config. Call with ignore-not-found=true to ignore resource not found errors during deletion.
+	$(KUSTOMIZE) build . | $(KUBECTL) delete --ignore-not-found=$(ignore-not-found) -f -
+
+##@ Tools
+
+## Location to install dependencies to
+TOOLBIN ?= $(CURDIR)/.cache/tools
+$(TOOLBIN):
+	mkdir -p $(TOOLBIN)
+
+## Tool Binaries
+CHAINSAW ?= $(TOOLBIN)/chainsaw
+CTLPTL ?= $(TOOLBIN)/ctlptl
+GOLANGCI_LINT ?= $(LOCALBIN)/golangci-lint
+KIND ?= $(TOOLBIN)/kind
+KUBECTL ?= kubectl ## Special case, we do not manage it via tools.go
+KUSTOMIZE ?= $(TOOLBIN)/kustomize
+
+## Tool Versions
+CHAINSAW_VERSION ?= $(shell grep 'github.com/kyverno/chainsaw ' ./hack/tools/go.mod | cut -d ' ' -f 2)
+CTLPTL_VERSION ?= $(shell grep 'github.com/tilt-dev/ctlptl ' ./hack/tools/go.mod | cut -d ' ' -f 2)
+GOLANGCI_LINT_VERSION ?= $(shell grep 'github.com/golangci/golangci-lint ' ./hack/tools/go.mod | cut -d ' ' -f 2)
+KIND_VERSION ?= $(shell grep 'sigs.k8s.io/kind ' ./hack/tools/go.mod | cut -d ' ' -f 2)
+KUSTOMIZE_VERSION ?= $(shell grep 'sigs.k8s.io/kustomize/kustomize/v5 ' ./hack/tools/go.mod | cut -d ' ' -f 2)
+
+.PHONY: chainsaw
+chainsaw: $(CHAINSAW)$(CHAINSAW_VERSION) ## Download chainsaw locally if necessary.
+$(CHAINSAW)$(CHAINSAW_VERSION): $(TOOLBIN)
+	$(call go-install-tool,$(CHAINSAW),github.com/kyverno/chainsaw,$(CHAINSAW_VERSION))
+
+.PHONY: ctlptl
+ctlptl: $(CTLPTL)$(CTLPTL_VERSION) ## Download ctlptl locally if necessary.
+$(CTLPTL)$(CTLPTL_VERSION): $(TOOLBIN)
+	$(call go-install-tool,$(CTLPTL),github.com/tilt-dev/ctlptl/cmd/ctlptl,$(CTLPTL_VERSION))
+
+.PHONY: golangci-lint
+golangci-lint: $(GOLANGCI_LINT)$(GOLANGCI_LINT_VERSION) ## Download golangci-lint locally if necessary.
+$(GOLANGCI_LINT)$(GOLANGCI_LINT_VERSION): $(LOCALBIN)
+	$(call go-install-tool,$(GOLANGCI_LINT),github.com/golangci/golangci-lint/cmd/golangci-lint,$(GOLANGCI_LINT_VERSION))
+
+.PHONY: kind
+kind: $(KIND)$(KIND_VERSION) ## Download kind locally if necessary.
+$(KIND)$(KIND_VERSION): $(TOOLBIN)
+	$(call go-install-tool,$(KIND),sigs.k8s.io/kind,$(KIND_VERSION))
+
+.PHONY: kustomize
+kustomize: $(KUSTOMIZE)$(KUSTOMIZE_VERSION) ## Download kustomize locally if necessary.
+$(KUSTOMIZE)$(KUSTOMIZE_VERSION): $(TOOLBIN)
+	$(call go-install-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v5,$(KUSTOMIZE_VERSION))
```

Comment:
Kustomize is part of `kubectl` for a while now. What's the benefit of having this as a separate tool?

Reply:
I don't think all of the functionality of Kustomize is present in Kubectl. I want to be able to do patches in the future via edit subcommand (e.g. ` $(KUSTOMIZE) edit set image controller=${IMG}`). This will either require `kubectl patch` or another tool for editing the YAML stream between `kubectl kustomize` and `kubectl apply`.